agent/engine/containermetadata/write_metadata.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package containermetadata
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/aws/amazon-ecs-agent/agent/api"
+	"github.com/cihub/seelog"
+)
+
+const (
+	metadataJoinSuffix = "metadata"
+	metadataFile       = "metadata.json"
+	readOnlyPerm       = 0644
+)
+
+// getTaskIDfromArn parses a task Arn and produces the task ID
+// A task Arn has format arn:aws:ecs:[region]:[account-id]:task/[task-id]
+// For a correctly formatted Arn we split it over ":" into 6 parts, the last part
+// containing the task-id which we extract by splitting it by "/".
+func getTaskIDfromArn(taskarn string) string {
+	colonSplitArn := strings.SplitN(taskarn, ":", 6)
```

Comment:
I've written a parser [here](https://godoc.org/github.com/samuelkarp/amazon-ecr-containerd-resolver/ecr/arn).  Need to figure out a good place for that code to live, but I think that's what you should use.